synth.py

```diff
@@ -40,31 +40,83 @@
  * limitations under the License.
  */
 """
+bad_license_header = """/\\*
+ \\* Copyright 2018 Google LLC
+ \\*
+ \\* Licensed under the Apache License, Version 2.0 \\(the "License"\\); you may not use this file except
+ \\* in compliance with the License. You may obtain a copy of the License at
+ \\*
+ \\* http://www.apache.org/licenses/LICENSE-2.0
+ \\*
+ \\* Unless required by applicable law or agreed to in writing, software distributed under the License
+ \\* is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ \\* or implied. See the License for the specific language governing permissions and limitations under
+ \\* the License.
+ \\*/
+"""
 
-service = 'datastore'
-versions = ['v1']
 
-for version in versions:
+def generate_client(service, version, proto_path=None, bazel_target=None, package=None):
   library = gapic.java_library(
       service=service,
       version=version,
-      bazel_target=f'//google/{service}/{version}:google-cloud-{service}-{version}-java',
-      proto_path=f'google/{service}/{version}',
+      proto_path=proto_path if proto_path else f'google/{service}/{version}',
+      bazel_target=bazel_target if bazel_target else f'//google/{service}/{version}:google-cloud-{service}-{version}-java',
   )
 
-  library = library / f'google-cloud-{service}-{version}-java'
+  library = library / f"google-cloud-{service}-{version}-java"
 
   s.replace(
       library / f'proto-google-cloud-{service}-{version}-java/src/**/*.java',
       protobuf_header,
       f'{license_header}{protobuf_header}'
   )
 
-  # Datastore only generates protos
-  s.copy(library / f'proto-google-cloud-{service}-{version}-java/src', f'proto-google-cloud-{service}-{version}/src')
+  if service == "datastore-admin":
```

Comment:
These fixes can be refactored, see https://github.com/googleapis/java-bigtable/blob/main/synth.py#L123-L125 (fix_proto_headers and fix_grpc_headers).